pkg/internal/controllers/certificaterequests.go

```diff
@@ -195,8 +195,8 @@ func (c *certificaterequests) reconcileStatusPatch(ctx context.Context, req ctrl
 		log.V(2).Info("approving request")
 		c.recorder.Event(cr, corev1.EventTypeNormal, "Approved", response.Message)
 
```

Comment:
Could we instead just return without doing a patch?

Reply:
I would prefer to not do it in this PR. That will require a lot of additional changes/refactoring. 

Not a bad idea, it will only require rewriting code and test.

Reply:
It seems like I was a bit quick to reject your suggestion, @inteon. After a walkthrough of this PR with @wallrj in today's stand-up, I realize that your suggestion makes a lot of sense to do now, and here it is: https://github.com/cert-manager/approver-policy/pull/303/commits/bb75196b1e98dcb078eb70888795c95529cfdde6. I still think we should bugfix the conditions patch - just in case the controller cache is stale.